### PR TITLE
add space image as driveItem and a trash facet for trash metadata 

### PR DESCRIPTION
**Pull request by micbar**
## Add space image to the graphAPI

related to https://github.com/owncloud/ocis/issues/2661

### Spec

The space Image will be represented as a driveItem on the drive object.

##### Space with image as special drive item
```json
{
  "value": [
    {
      "id": "string",
      "createdBy": {},
      "createdDateTime": "2021-11-24T16:39:06.730Z",
      "description": "string",
      "eTag": "string",
      "lastModifiedBy": {},
      "lastModifiedDateTime": "2021-11-24T16:39:06.730Z",
      "name": "Marketing",
      "webUrl": "string",
      "createdByUser": {},
      "lastModifiedByUser": {},
      "driveType": "project",
      "owner": {},
      "quota": {
        "deleted": 0,
        "remaining": 0,
        "state": "string",
        "total": 0,
        "used": 0
      },
      "special": [{
        "id": "string",
        "createdDateTime": "2021-11-24T16:39:06.730Z",
        "description": "string",
        "eTag": "string",
        "lastModifiedDateTime": "2021-11-24T16:39:06.730Z",
        "name": "spaceImage",
        "webUrl": "string",
        "cTag": "string",
        "deleted": {
          "state": "string"
        },
        "file": {
          "hashes": {
            "crc32Hash": "string",
            "quickXorHash": "string",
            "sha1Hash": "string",
            "sha256Hash": "string"
          },
          "mimeType": "string",
          "processingMetadata": true
        },
        "fileSystemInfo": {
          "createdDateTime": "2021-11-24T16:39:06.730Z",
          "lastAccessedDateTime": "2021-11-24T16:39:06.730Z",
          "lastModifiedDateTime": "2021-11-24T16:39:06.730Z"
        },
        "image": {
          "height": 2147483647,
          "width": 2147483647
        },
      }]
    }
  ]
}
```

## To be discussed

- This only defines the payload of the api
- We need to define which attributes are part of the `GET /me/drives/` request by default and which we will respond when providing the `$extend=image` parameter

### CODE REVIEW

**Comment by refs** [comment]
should we start considering payload sized with this change?

**Comment by micbar** [comment]
we should be smart here. The clients want do decide when they request a response with the file content. The image will be cached. For the following requests the clients will just check the ctag which will be resource friendly. Only when the content of the file has changed, the request will extent the image content.

**Comment by butonic** [comment]
1. driveItems have an image facet: https://docs.microsoft.com/en-us/graph/api/resources/image?view=graph-rest-1.0 the naming might be confusing .
2. Clients might want to have different sizes: https://docs.microsoft.com/en-us/graph/api/driveitem-list-thumbnails?view=graph-rest-1.0&tabs=http so fetching a custom size for the space image should be done with the thumbnails api.

**Review by felix-schwarz — APPROVED** [review]

**Review by michaelstingl — APPROVED** [review]

**Comment by micbar** [comment]
let us not merge yet, i will address @butonic s feedback first

**Comment by micbar** [comment]
@butonic @felix-schwarz @TheOneRing @michaelstingl Let me add the thumbnail parts.

**Comment by refs** [comment]
@micbar is [this](https://github.com/owncloud/libre-graph-api/pull/9/commits/d967b9e1aa0afaac16926f55b424d5f727fdfa6e) relevant for this PR?

**Comment by micbar** [comment]
> @micbar is [this](https://github.com/owncloud/libre-graph-api/pull/9/commits/d967b9e1aa0afaac16926f55b424d5f727fdfa6e) relevant for this PR?

Could be a new one, yes.

File: api/openapi-spec/v0.0.yaml [hunk 1]
```diff
@@ -971,6 +975,8 @@ components:
           readOnly: true
         root:
           $ref: '#/components/schemas/driveItem'
+        spaceImage:
```
Comment (butonic) [inline api/openapi-spec/v0.0.yaml:978]: we want to model this as a `image` driveItem in the drive `special` drive Item collection

**Comment by refs** [comment]
> > @micbar is [this](https://github.com/owncloud/libre-graph-api/pull/9/commits/d967b9e1aa0afaac16926f55b424d5f727fdfa6e) relevant for this PR?
> 
> Could be a new one, yes.

you could augment the scope of the PR to include it, but mention it somewhere nevertheless for future generations 👍

**Review by butonic — APPROVED** [review]

**Pull request merged**